### PR TITLE
feat: add I64Const32S/I64Const32U opcodes to emit i64.const as one instruction (FLU-1182)

### PR DESCRIPTION
**Pull request by dmitry123**
## What

`op_i64_const` previously lowered every `i64.const` into two `I32Const` instructions (2 instrs / 16 encoded bytes). This adds two fused opcodes that carry a single 32-bit immediate and push both limbs in one step, producing the exact same stack layout (`lo` pushed first, then `hi`, top = hi):

- `I64Const32S(value)` (code 88) — high limb = sign-extension of the immediate
- `I64Const32U(value)` (code 89) — high limb = 0

Constants that genuinely need all 64 bits keep the two-`I32Const` lowering. Measured on the nitro asset, 59.8% of `i64.const` values are sign-extendable, 36.6% have a zero high limb, and only 3.6% (579 of 16,178) need the fallback — so 96.4% of uses become 1 instr / 8 encoded bytes.

**`Opcode` stays 8 bytes in memory** (single-register loads, no cache-footprint regression for the rest of the bytecode); `test_opcode_size` pins this. Circuit-wise both variants are limb-local: the high limb is a trivial function of the immediate (arithmetic shift or zero), the same provability class as merged `I32Const` rows, matching the `I32Add64`/`I32Mul64`/`I32And64`/`I32Sub64` precedent. The encoding stays fully 32-bit-aligned (u32 code + u32 immediate).

## Changes

- `src/types/opcode.rs`: new `I64Const32S`/`I64Const32U` variants (codes 88/89, after `I32Sub64` from #187), `Display` arms, round-trip encode/decode test, updated code-value test, `test_opcode_size` pins `size_of::<Opcode>() == 8`
- `src/isa/memory.rs`: `op_i64_const` picks S / U / two-`I32Const` fallback; `MSH_I64_CONST` stays 2; `f64.const` benefits via the same path
- `src/vm/executor.rs` + `src/vm/executor/stack.rs`: dispatch + `visit_i64_const32_s`/`_u` (2 pushes, 1 ip step)
- `docs/opcodes.md`: catalog rows for codes 88–89

## Measured impact

Same config on `tests/assets`, current devel (3073baf6, incl. #185–#189) vs this branch:

| asset | instrs | encoded bytes |
|---|---|---|
| nitro-verifier | 289,921 → 274,321 (**−15,600, −5.4%**) | 2,639,349 → 2,514,549 (**−124,800, −4.7%**) |
| secp256k1 | 26,535 → 26,077 | 2,355,093 → 2,351,429 |
| panic | 3,014 → 3,004 | 64,047 → 63,967 |

Encoded saving is −8 bytes per fitting use (16 → 8), matching the FLU-1182 estimate of −16K instrs / −5% count on nitro.

## Verification

- Full `cargo test` suite passes on top of current devel (incl. `tests/snippets.rs` `test_i64_const`, whose edge-case constants exercise all three lowerings through the VM, and the wasmtime differential strategy tests in `tests/fuzz.rs`)
- `cargo test --release --test fluentbase -- --ignored` (real nitro attestation verification workload) passes
- New encode/decode round-trip test pins the wire format: `[code, imm]` little-endian, 8 bytes
- `cargo clippy --all-targets` clean

Linear: [FLU-1182](https://linear.app/fluentlabs-xyz/issue/FLU-1182)

### CODE REVIEW

**Review by d1r1 — APPROVED** [review]

File: src/types/opcode.rs [hunk 1]
```diff
@@ -186,7 +186,13 @@ define_opcode_enum! {
...
     I32Or64 => 85u32,
     I32Xor64 => 86u32,
-    I32Sub64 => 87u32,
+    // push a 64-bit constant whose value fits the single 32-bit immediate: the immediate is
+    // pushed as the low limb, then the derived high limb (top of stack = hi), matching the
+    // layout two consecutive `I32Const` pushes produce; `S` sign-extends the immediate into
+    // the high limb, `U` zero-extends it. Constants needing all 64 bits keep the two-`I32Const`
+    // lowering, so `Opcode` stays 8 bytes in memory
+    I64Const32S(value: UntypedValue) => 87u32,
+    I64Const32U(value: UntypedValue) => 88u32,
```
[the remark this answers is not in the document]
Comment (dmitry123) [reply src/types/opcode.rs]: Fixed in e7174d12: aux_value() now returns value.to_bits() for both variants and both are classified as nullary, with regression cases in aux_value_surfaces_every_immediate and opcode_class_predicates. is_const_instruction() intentionally still excludes them: it drives the tracer's single-slot write recording (record_mw records only the top-of-stack value), and these opcodes write two slots — marking them const would record the high limb and silently drop the low-limb write. They follow the same rule as the other two-slot writers (I32Add64/I32Sub64/I32And64/...), which trace backends handle explicitly.

**Review by djadjka — APPROVED** [review]
